src/main/asciidoc/jpa.adoc

```diff
@@ -685,6 +685,22 @@ The escape character used can be configured by setting the `escapeCharacter` of
 Note that the method `escape(String)` available in the SpEL context will only escape the SQL and JPQL standard wildcards `_` and `%`.
 If the underlying database or the JPA implementation supports additional wildcards these will not get escaped.
 
+[[jpa.query.other-methods]]
+=== Other Methods
+
+Spring Data JPA offers many ways to build queries.
+But sometimes, your query may simply be too complicated for the techniques offered.
+In that situation, consider:
+
+* If you haven't already, simply write the query yourself using <<jpa.query-methods.at-query,`@Query`>>.
+* Sometimes, you can put your query inside the database use Spring Data JPA's <<jpa.stored-procedures,`@StoredProcedure` annotation>>, or if it's a database function use the <<jpa.query-methods.at-query,`@Query` annotation>> and invoke it with a `CALL`.
```

Comment:
Let's mention stored procedures after recommending `JdbcTemplate` because it creates a tighter coupling with the database requirements.